core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -560,6 +560,9 @@ private <K, V> Cache<K, V> internalGetCache(String cacheName) {
 
    @Override
    public boolean cacheExists(String cacheName) {
+      if (getCacheBlockingCheck != null) {
```

Comment:
This has to be on the isRunning method. Existance is not the same thing as the cache running. Without a isRunning check to verify the cache is actually running, the getCache method may block unexpectedly. That is the point of this method for tests.

Reply:
The blocking thread I mentioned before was because of this. The `DefaultCacheManager` calls `BlockHoundUtil.pretendBlock()`. I updated it to have the same execution as before but added a topology check. With this check earlier, we don't have to initiate the operation to find the missing members then.

Reply:
Looks like this wasn't changed?

Reply:
Sorry, I missed a commit. Should be correct now.